index.js

```diff
@@ -793,25 +797,28 @@ Command.prototype.normalize = function(args) {
 Command.prototype.parseArgs = function(args, unknown) {
   var name;
 
+  if (this.listeners('program-action').length && this.listeners('command:*').length) {
+    console.error("Can't have listeners for both command:* and for the main program");
+    this._exit(1);
+  }
+
   if (args.length) {
     name = args[0];
     if (this.listeners('command:' + name).length) {
       this.emit('command:' + args.shift(), args, unknown);
-    } else {
```

Comment:
The old behaviour was emit `command:foo` if defined, else emit `command:*`. There could be multiple listeners on `command:*`.

I think the backwards compatible change is: emit `command:foo` if defined, else emit `command:*` and also emit `program-action` if defined. (Don't actually need to test for `program-action` being defined but it feels wrong to dispatch that event in subcommands.)

So I suggest:

```
} else {
  if (this.listeners('program-action').length) {
    this.emit('program-action', args, unknown);
  }
  this.emit('command:*', args, unknown);
}
```

Reply:
Hmmm.. the current logic would call `command:*` if there are no matches for the provided command name and there’s at least one * listener. 

That does seem to back up the current behavior, right?

Reply:
I didn't quite understand your comment. Does my suggestion make more sense in the light of the separate comments about calling both being backwards compatible? I suspect we had different expectations about calling one/both.

Otherwise, comment again as I missed the point! (Thanks.)

Reply:
This discussion is helping my understanding. I am updating my suggestion to be more self-consistent. The two cases are symmetrical so should be treated the same. Simply:

```
} else {
  this.emit('program-action', args, unknown);
  this.emit('command:*', args, unknown);
}
```

Reply:
👍 I updated this as suggested